staging/src/k8s.io/apiserver/pkg/util/peerproxy/peerproxy.go

```diff
@@ -17,51 +17,157 @@ limitations under the License.
+// This method is used for an alpha feature UnknownVersionInteroperabilityProxy
+// and is subject to future modifications.
+func NewPeerProxyHandler(
 	serverId string,
+	identityLeaseLabelSelector string,
+	leaseInformer coordinationv1informers.LeaseInformer,
 	reconciler reconcilers.PeerEndpointLeaseReconciler,
-	serializer runtime.NegotiatedSerializer) *peerProxyHandler {
+	ser runtime.NegotiatedSerializer,
+	loopbackClientConfig *rest.Config,
+	proxyClientConfig *transport.Config,
+) (*peerProxyHandler, error) {
 	h := &peerProxyHandler{
-		name:                  "PeerProxyHandler",
-		storageversionManager: svm,
-		proxyTransport:        proxyTransport,
-		svMap:                 sync.Map{},
-		serverId:              serverId,
-		reconciler:            reconciler,
-		serializer:            serializer,
+		name:                             "PeerProxyHandler",
+		serverID:                         serverId,
+		reconciler:                       reconciler,
+		serializer:                       ser,
+		localDiscoveryInfoCache:          atomic.Value{},
+		localDiscoveryCacheTicker:        time.NewTicker(localDiscoveryRefreshInterval),
+		localDiscoveryInfoCachePopulated: make(chan struct{}),
+		peerDiscoveryInfoCache:           atomic.Value{},
+		peerLeaseQueue: workqueue.NewTypedRateLimitingQueueWithConfig(
+			workqueue.DefaultTypedControllerRateLimiter[string](),
+			workqueue.TypedRateLimitingQueueConfig[string]{
+				Name: controllerName,
+			}),
+		apiserverIdentityInformer: leaseInformer,
+	}
+
+	if parts := strings.Split(identityLeaseLabelSelector, "="); len(parts) != 2 {
+		return nil, fmt.Errorf("invalid identityLeaseLabelSelector provided, must be of the form key=value, received: %v", identityLeaseLabelSelector)
+	}
+	selector, err := labels.Parse(identityLeaseLabelSelector)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse label selector: %w", err)
+	}
+	h.identityLeaseLabelSelector = selector
+
+	discoveryScheme := runtime.NewScheme()
+	utilruntime.Must(apidiscoveryv2.AddToScheme(discoveryScheme))
+	h.discoverySerializer = serializer.NewCodecFactory(discoveryScheme)
+
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(loopbackClientConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error creating discovery client: %w", err)
+	}
+	h.discoveryClient = discoveryClient
+	h.localDiscoveryInfoCache.Store(map[schema.GroupVersionResource]bool{})
+	h.peerDiscoveryInfoCache.Store(map[string]map[schema.GroupVersionResource]bool{})
+
+	proxyTransport, err := transport.New(proxyClientConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create proxy transport: %w", err)
 	}
-	svi := informerFactory.Internal().V1alpha1().StorageVersions()
-	h.storageversionInformer = svi.Informer()
+	h.proxyTransport = proxyTransport
 
-	svi.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+	peerDiscoveryRegistration, err := h.apiserverIdentityInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
sorry I know this section has gone through a bunch of refactoring and just want to double check: Do events coming from this informer already filter for identity leases? If not then we should check the label in `isValidPeerIdentityLease` so we don't add random node leases to the queue right?

Reply:
Added